packages/web/src/components/RedwoodProvider.js

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'react'
+
+import { GraphQLProvider, createGraphQLClient } from 'src/graphql'
+
+const useAuthStub = () => ({ loading: false, authenticated: false })
+
+/**
+ * Redwood's Provider ties together our authentication and Apollo's GraphQL Provider
+ * when you pass the `useAuth` hook from `@redwoodjs/auth`
+ * The authentication token is added to the headers of the GraphQL client.
+ */
+const RedwoodProvider = ({ useAuth = useAuthStub, children }) => {
+  const { loading, authenticated, getToken, type } = useAuth()
+  const [authToken, setAuthToken] = useState()
+
+  useEffect(() => {
+    if (authenticated) {
+      getToken().then((token) => setAuthToken(token))
+    }
+  }, [authenticated, getToken])
+
+  // This really sucks because rendering is completely blocked whilst we're
+  // restoring authentication.
+  if (loading) {
```

Comment:
Hmm interesting. What if we wrapped the restoreAuth call in a `requestIdleCallback` here? https://github.com/redwoodjs/redwood/pull/497/files#diff-750ebfc0b1cca76d488951096093372dR55

Reply:
Right now, I'm OK with shipping this, but would love to improve it directly after it has landed - I'll leave a note!

The reason why I'm ok with it is because:
1. Both Auth0 and Netlify allow you to cache the state locally, so hydration shouldn't induce that much latency.
2. I'm trying to figure out what the expectation of a developer and user would be... If you come to the site initially and it's busy figuring out if you're authenticated then we shouldn't really show you anything/ kick off queries/ asynchronously download pages until that's been determined.

I don't actually have much context for what `requestIdleCallback` would achieve here (It's new to me in this context), but I would love to investigate that further.